src/main/extension/extension_load.cpp

```diff
@@ -119,15 +152,66 @@ bool ExtensionHelper::TryInitialLoad(DBConfig &config, FileSystem &fs, const str
 		error = StringUtil::Format("Extension \"%s\" not found.\n%s", filename, message);
 		return false;
 	}
-	if (!config.options.allow_unsigned_extensions) {
-		auto handle = fs.OpenFile(filename, FileFlags::FILE_FLAGS_READ);
 
-		// signature is the last 256 bytes of the file
+	string metadata_segment;
+	metadata_segment.resize(512);
+
+	const std::string engine_version = std::string(GetVersionDirectoryName());
+	const std::string engine_platform = std::string(DuckDB::Platform());
+
+	auto handle = fs.OpenFile(filename, FileFlags::FILE_FLAGS_READ);
+
+	idx_t file_size = handle->GetFileSize();
+
+	if (file_size < 1024) {
```

Comment:
Should this be 512? And maybe move this to a `constexpr` instead of a magic number